packages/cadl-ranch-specs/http/spread/main.tsp

```diff
@@ -0,0 +1,158 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for the spread operator.")
+@supportedBy("dpg")
+@scenarioService("/spread")
+namespace Spread;
+
+@doc("This is a simple model.")
+model ModelBodyParameter {
+  name: string;
+}
+
+@scenario
+@scenarioDoc("""
+Test case for spread named model. Should generate request body model named ModelBodyParameter.
+
+Expected request body:
+```json
+{ "name": "foo" }
+```
+""")
+@route("/model/request-body")
+@put
+op spreadModelAsRequestBody(...ModelBodyParameter): OkResponse;
+
+@doc("This is a simple model including path parameter.")
+model ModelRequestParamter {
+  @path
+  id: string;
+
+  @header
+  `x-ms-test-header`: string;
+  name: string;
+}
+
+@scenario
+@scenarioDoc("""
+Test case for spread named model with path and header parameter. 
+
+Expected path parameter: id=1
+Expected header parameter: x-ms-test-header="bar"
+Expected request body:
+```json
+{ "name": "foo" }
+```
+""")
+@route("/model/request-parameter/{id}")
+@put
+op spreadModelAsRequestParameter(...ModelRequestParamter): OkResponse;
+
+alias AliasBodyParameter = {
+  name: string;
+};
+
+@scenario
+@scenarioDoc("""
+Test case for spread alias. 
```

Comment:
Do we need to say that no model should be generated?

Reply:
@lmazuel I'd like to hear your feedback here. For all spread cases, we are pretty sure what kind of http payload would be look like. But the questions for whether we are supposed to generate a model or should not generate any model currently it may be lack of consistency across language. Do you think  we should keep consistent and we need to specify in cadl-ranch?

Reply:
We should never generate a public model from an alias. There is no exception to that rule, as alias is just a way to simplify some syntax, and removing the alias later is NOT a breaking change in TypeSpec. If you rely on it for model generation, bad things will happen.

I can understand that for serialization purposes some language needs to generate a private internal model. Since it would be private, the name of that model doesn't matter.